langchain/vectorstores/annoy.py

```diff
@@ -310,9 +310,13 @@ def __from(
             metadata = metadatas[i] if metadatas else {}
             documents.append(Document(page_content=text, metadata=metadata))
         index_to_id = {i: str(uuid.uuid4()) for i in range(len(documents))}
```

Comment:
still not sure i understand how index_to_id could have different length than documents in this case

Reply:
it is possible when you pass ids while calling `from_texts` or `from_embeddings` methods in FAISS but I noticed in annoy `ids` are always created from inside `_from` method and not passed by the user, so there this will not happen.

I figured this issue when I passed list of chunks of an article with name of the article as `ids`. Index were created successfully, but when I queried, I wasn't able to get the relevant chunk of the article.

Reply:
@baskaryan could you review the new patch again?